dumper/app/src/main/java/com/google/edwmigration/dumper/application/dumper/connector/teradata/AbstractTeradataConnector.java

```diff
@@ -132,17 +132,17 @@ public TeradataJdbcSelectTask(String targetPath, TaskCategory category, String s
     @Nonnull
     private ResultSetExtractor<Summary> newCountedResultSetExtractor(
         @Nonnull ByteSink sink, @Nonnull Connection connection) throws SQLException {
-      long count = -1;
       if (sqlCount != null) {
         // It's a lot of infrastructure, but we don't have to write it.
         RowMapper<Long> rowMapper = new SingleColumnRowMapper<>(Long.class);
-        ResultSetExtractor<List<Long>> resultSetExtractor =
-            new RowMapperResultSetExtractor<>(rowMapper);
-        List<Long> results = doSelect(connection, resultSetExtractor, sqlCount);
-        Long result = DataAccessUtils.nullableSingleResult(results);
-        if (result != null) count = result;
+        ResultSetExtractor<List<Long>> extractor = new RowMapperResultSetExtractor<>(rowMapper);
+        List<Long> results = doSelect(connection, extractor, sqlCount);
+        Optional<Long> count = Optional.ofNullable(nullableSingleResult(results));
+        if (count.isPresent() && count.get() >= 0) {
```

Comment:
The use of Optional here over `@Nullable Long` seems to add, rather than remove complexity.

Reply:
True, I changed it back to `@Nullable Long`.

Reply:
I ended up reviewing the guidelines, and they say "avoid null where unnecessary" and "use optional where appropriate" and the common interpretation seems to be "use optional where possible" which is not, I think, what was intended.  😁

Reply:
I'm not sure about this guideline in this context, I would read it more as an API design principle.

Before this change we had two variables, `long count` and `Long result`, both of which meant the same thing, except `result == null` was equivalent to `count == -1`.
At some point during the change, it was convenient to introduce Optional for `count`, but in the end both `result` and the Optional wrapper became unnecessary.